rails-commenting.rb

```diff
@@ -5,27 +5,29 @@
 
 # app/controller/blog_posts_controller.rb
 
-# 1)
+# $ rails g controller blog_posts
+
+# 1) After we generate a controller, a class is created inside the controller file for Blog Posts. The controller class inherits from ApplicationController. 
 class BlogPostsController < ApplicationController
   def index
-    # 2)
+    # 2) Defined a method called index that creates an instance variable of @posts that gets all of the Blog Posts that have been created in the database.
     @posts = BlogPost.all
   end
 
   def show
-    # 3)
+    # 3) Defined a show method that creates an instance  variable of @blog, which will show a specific blog post when searched for in the URL with its assigned id. Example: localhost:3000/blog_posts/3
     @post = BlogPost.find(params[:id])
   end
 
-  # 4)
+  # 4) The new method creates an object instance that is empty. If a user goes to clients/new in the application to add a new blog, Rails will create an instance of Blog_Posts and run the new method. Handles only get requests. Does not create/post anything. Represents a midway point between Create and Read. Rails convention that displays a form to the user.
```

Comment:
Totally. I think I would also want to mention why its empty and why that's OK,  a bit more, but you got it!